### DISCUSSION TITLE
멀티에서 텔레포트 위치를 여러곳 지정하려면 어떻게 하나요?

### DISCUSSION THREAD

**Discussion by jgw1413** [naverz/zepeto-studio-kor#1119] [5. Q&A]
멀티에서 텔레포트 기능 구현중인데 클라이언트 캐릭터 State가 Teleport일때 동기화 시켜주고 있습니다.

근데 텔레포트 해야할 위치가 여러개인데 클라이언트 캐릭터가 어떤곳으로 텔레포트 했는지 어떻게 알 수 있을까요?

**Comment by smlee2952** [comment]
State를 이용하지 마시고, room.Send()를 통해 텔레포트 메세지에 좌표데이터를 같이 보내주면 될 것 같습니다.

**Reply by smlee2952** [reply]
https://studio.zepeto.me/guides/multiplay-room-message

이 부분을 참고하시어, 텔레포트 시 전송할 메세지 정의 후
해당 메세지를 서버에 전송 시 텔레포트 할 좌표 데이터를 같이 넣어 각각의 클라이언트들에게 전달해주면
될 것 같습니다.

**Reply by jgw1413** [reply]
답변 감사합니다! 말씀해주신데로 서버로 메세지 전송 후 처리해주고 있는데 teleport 할때 클라이언트 캐릭터가 순간이동을 하지않고 빠르게 이동하는 문제가 있더라구요 ㅠㅠ..

**Reply by smlee2952** [reply]
제페토 예제코드에서는 다른 플레이어들의 위치 정보를 받으면 해당 좌표로 뛰어가게 되어 있습니다.

다른 클라이언트의 텔레포트 메세지 수신 시 텔레포트 함수를 호출하시나요?

**Reply by smlee2952** [reply]
https://github.com/naverz/zepeto-studio-kor/discussions/1090

비슷한 사례입니다ㅎㅎ

**Reply by jgw1413** [reply]
현재 그 사례 참고하여 구현했는데 말씀처럼 제페토 멀티샘플 이용해서 그럴수도 있을거같아요 이부분을 한번 확인해보겠습니다 감사합니다!

**Comment by YSHFM** [comment]
저같은 경우는 좌표값을 항시 서버로 갱신해주고 있고 transform을 직접 이동시킴으로 텔레포트 작업을 사용하고 있습니다. 트랜스폼을 클라이언트에서 이동시키면 어차피 좌표를 갱신하고 있기 때문에 동기화가 된 것으로 보입니다.
